gtsfm/loader/loader_base.py

```diff
@@ -81,6 +95,73 @@ def is_valid_pair(self, idx1: int, idx2: int) -> bool:
             validation result.
         """
 
+    def get_image(self, index: int) -> Image:
+        """Get the image at the given index, satisfying a maximum image resolution constraint.
+
+        Determine how the camera intrinsics and images should be jointly rescaled based on desired img. resolution.
+        Each loader implementation should set a `_max_resolution` attribute.
+
+        Args:
+            index: the index to fetch.
+
+        Raises:
+            IndexError: if an out-of-bounds image index is requested.
+
+        Returns:
+            Image: the image at the query index. It will be resized to satisfy the maximum
+                allowed loader image resolution if the full-resolution images for a dataset
+                are too large.
+        """
+        # no downsampling may be required, in which case target_h and target_w will be identical
+        # to the full res height & width.
+        img_full_res = self.get_image_full_res(index)
+        (
+            _,
+            _,
+            target_h,
+            target_w,
+        ) = img_utils.get_downsampling_factor_per_axis(img_full_res.height, img_full_res.width, self._max_resolution)
+        logger.info(
+            "Image %d resized from (H,W)=(%d,%d) -> (%d,%d)",
+            index,
+            img_full_res.height,
+            img_full_res.width,
+            target_h,
+            target_w,
+        )
+        resized_img = img_utils.resize_image(img_full_res, new_height=target_h, new_width=target_w)
+        return resized_img
+
+    def get_camera_intrinsics(self, index: int) -> Cal3Bundler:
+        """Get the camera intrinsics at the given index, for a possibly resized image.
+
+        Determine how the camera intrinsics and images should be jointly rescaled based on desired img. resolution.
+        Each loader implementation should set a `_max_resolution` attribute.
+
+        Args:
+            the index to fetch.
+
+        Returns:
+            intrinsics for the given camera.
+        """
+        intrinsics_full_res = self.get_camera_intrinsics_full_res(index)
+        if intrinsics_full_res is None:
+            raise ValueError(f"No intrinsics found for index {index}.")
+
+        img_full_res = self.get_image_full_res(index)
+        # no downsampling may be required, in which case scale_u and scale_v will be 1.0
+        scale_u, scale_v, _, _ = img_utils.get_downsampling_factor_per_axis(
+            img_full_res.height, img_full_res.width, self._max_resolution
+        )
+        rescaled_intrinsics = Cal3Bundler(
+            fx=intrinsics_full_res.fx() * scale_u,
+            k1=0.0,
+            k2=0.0,
```

Comment:
why are k1 and k2 zeros? Can we not obtain this factor from simple scaling of the original values too?

Reply:
So currently we have k1 and k2 hard coded in the different 

https://github.com/borglab/gtsfm/blob/master/gtsfm/common/image.py#L68
https://github.com/borglab/gtsfm/blob/master/gtsfm/utils/io.py#L189

I opened an issue https://github.com/borglab/gtsfm/issues/301. Since they are coefficients of polynomials, a linear rescaling shouldn't be sufficient, though.